src/main/java/bio/terra/app/configuration/ApplicationConfiguration.java

```diff
@@ -432,8 +433,11 @@ public RestTemplate restTemplate(RestTemplateBuilder builder) {
   @Bean("tdrServiceAccountEmail")
   public @Nullable String tdrServiceAccountEmail() throws IOException {
     GoogleCredentials defaultCredentials = GoogleCredentials.getApplicationDefault();
-    if (defaultCredentials instanceof ServiceAccountCredentials) {
-      return ((ServiceAccountCredentials) defaultCredentials).getClientEmail();
+    if (defaultCredentials instanceof ServiceAccountCredentials serviceAccountCredentials) {
+      return serviceAccountCredentials.getClientEmail();
+    }
+    if (defaultCredentials instanceof ExternalAccountCredentials externalAccountCredentials) {
+      return externalAccountCredentials.getServiceAccountEmail();
```

Comment:
We would have to handle three cases:
1) service account for service account running in environments
2) ExternalAccountCredentials for workload identity
3) Impersonated credentials for local run if we auth while impersonating the TDR SA